pallets/price-oracle/src/lib.rs

```diff
@@ -0,0 +1,231 @@
+// This file is part of Basilisk-node.
+
+// Copyright (C) 2020-2021  Intergalactic, Limited (GIB).
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use frame_support::pallet_prelude::Weight;
+use frame_support::sp_runtime::traits::Zero;
+use primitives::{
+	asset::AssetPair,
+	traits::{AMMHandlers, AMMTransfer},
+	AssetId, Balance,
+};
+use sp_std::marker::PhantomData;
+use sp_std::prelude::*;
+
+#[cfg(test)]
+mod mock;
+
+#[cfg(test)]
+mod tests;
+
+mod types;
+pub use types::*;
+
+pub mod weights;
+use weights::WeightInfo;
+
+mod benchmarking; // TODO: rebenchmark
+
+// Re-export pallet items so that they can be accessed from the crate namespace.
+pub use pallet::*;
+
+pub type AssetPairId = Vec<u8>;
```

Comment:
This type looks bit strange to me. 

Perhaps it just the name.


A description would be good. I can;t tell what it should represent.